### PR TITLE
cifar-pytorch

### PR DESCRIPTION
**Pull request by chandramouli-sastry**
* Contains code for cifar-pytorch and edits the train and validation split sizes. 
* I've also added parameter shape-tests in the corresponding test-files. 
* The code is largely based off of the imagenet-pytorch repo.

### CODE REVIEW

**Comment by chandramouli-sastry** [comment]
recheck

File: algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+from algorithmic_efficiency import data_utils
+from algorithmic_efficiency import param_utils
+from algorithmic_efficiency import spec
+import algorithmic_efficiency.random_utils as prng
+from algorithmic_efficiency.workloads.cifar.workload import BaseCifarWorkload
+from algorithmic_efficiency.workloads.imagenet.imagenet_pytorch.models import \
+    resnet18
+
+PYTORCH_DDP = 'LOCAL_RANK' in os.environ
```
Comment (runame) [inline algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py:25]: Could you change `PYTORCH_DDP` to `USE_PYTORCH_DDP`?

File: algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        transforms.RandomResizedCrop(
+            self.center_crop_size,
+            scale=self.scale_ratio_range,
+            ratio=self.aspect_ratio_range),
+        transforms.RandomHorizontalFlip(),
+        normalize
+    ])
+
+    dataset = CIFAR10(root=data_dir,
+                train=split in ["train","eval_train","validation"],
```
Comment (runame) [inline algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py:88]: Nit: it would be great if you could consistently use single instead of double quotation marks.

**Review by runame — APPROVED** [review]
LGTM! Did you already try to run the submission with the current setup (optimizer, lr schedule, hyperparameters)?

**Review by znado — APPROVED** [review]
LGTM!

File: algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py (hunk 1 above)
Comment (chandramouli-sastry) [reply algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py]: Fixed!

File: algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py (hunk 2 above)
Comment (chandramouli-sastry) [reply algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py]: (And yeah, I did test-run the submission and it is very similar to the imagenet submission code)

Comment (runame) [reply algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py]: Cool, what kind of validation performance did you get and after what time?

Comment (chandramouli-sastry) [reply algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py]: Both PyTorch and Jax models get ~86% accuracy after 60k steps. Few notes:

- Jax trains on entire train partition (50k) whereas pytorch trains on 40k. 
- Jax is able to get in 60k steps in 600 seconds whereas pytorch takes 2400 seconds (i.e. 4x slower) -- I set the number of workers in pytorch dataloader to 4 to get these results... the committed code has `num_workers=0`

Comment (fsschneider) [reply algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py]: 4x slower is a lot. Do you have an idea where that slowdown is coming from?

**Comment by fsschneider** [comment]
@chandramouli-sastry can you fix the merge conflicts?
Then I can merge it, the CLA bot now succeeds :)
